retrieve/libp2ptransfermanager.go

```diff
@@ -0,0 +1,432 @@
+package retrieve
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/filecoin-project/boost/transport/httptransport"
+	boosttypes "github.com/filecoin-project/boost/transport/types"
+	datatransfer "github.com/filecoin-project/go-data-transfer"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+// libp2pCarServer is mocked out by the tests
+type libp2pCarServer interface {
+	ID() peer.ID
+	Start(ctx context.Context) error
+	Get(id string) (*httptransport.Libp2pTransfer, error)
+	Subscribe(f httptransport.EventListenerFn) httptransport.UnsubFn
+	CancelTransfer(ctx context.Context, id string) (*boosttypes.TransferState, error)
+	Matching(f httptransport.MatchFn) ([]*httptransport.Libp2pTransfer, error)
+}
+
+// libp2pTransferManager watches events for a libp2p data transfer
+type libp2pTransferManager struct {
+	dtServer  libp2pCarServer
+	authDB    *httptransport.AuthTokenDB
+	dtds      datastore.Batching
+	opts      libp2pTransferManagerOpts
+	ctx       context.Context
+	cancelCtx context.CancelFunc
+	ticker    *time.Ticker
+
+	listenerLk sync.Mutex
+	listener   func(uint, ChannelState)
+	unsub      func()
+}
+
+type libp2pTransferManagerOpts struct {
+	xferTimeout     time.Duration
+	authCheckPeriod time.Duration
+}
+
+func newLibp2pTransferManager(dtServer libp2pCarServer, ds datastore.Batching, authDB *httptransport.AuthTokenDB, opts libp2pTransferManagerOpts) *libp2pTransferManager {
```

Comment:
I don't think we need libp2pTransferManager - it's used to manage incoming requests for data in storage deals